modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -627,13 +629,43 @@ private void parseNullSafe(ExtensionParameter extensionParameter, ParameterDecla
         @Override
         protected void visitBasicType(MetadataType metadataType) {
           throw new IllegalParameterModelDefinitionException(format("Parameter '%s' is annotated with '@%s' but is of type '%s'. That annotation can only be "
-              + "used with complex types (Pojos, Lists, Maps)",
+              + "used with complex instantiable types (Pojos, Lists, Maps)",
                                                                     extensionParameter.getName(), NullSafe.class.getSimpleName(),
                                                                     extensionParameter.getType().getName()));
         }
       });
 
-      parameter.withModelProperty(new NullSafeModelProperty());
+      Class<?> defaultType = extensionParameter.getAnnotation(NullSafe.class).get().defaultImplementingType();
+
+
+      if (isInstantiable(parameter.getDeclaration().getType()) && !defaultType.isAssignableFrom(Object.class)) {
```

Comment:
- the defaultType _always_ needed, the wrapper always spects the model property to know what's the type that has to be instantiated.
- if the defaultType is not specified, then we fallback to the parameter's type, just like we did before.

both of this things can be seen in the statement below the commented lines: 

`MetadataType nullSafeType = defaultType.isAssignableFrom(Object.class)
          ? parameter.getDeclaration().getType() : typeLoader.load(defaultType);`